pkg/parse/root_test.go

```diff
@@ -107,46 +78,54 @@ func gitSpec(repo string, auth configsync.AuthType) core.MetaMutator {
 
 func TestRoot_Parse(t *testing.T) {
 	testCases := []struct {
-		name              string
-		format            filesystem.SourceFormat
-		namespaceStrategy configsync.NamespaceStrategy
-		existingObjects   []client.Object
-		parsed            []ast.FileObject
-		want              []ast.FileObject
+		name                string
+		format              filesystem.SourceFormat
+		namespaceStrategy   configsync.NamespaceStrategy
+		existingObjects     []client.Object
+		expectedObjsToApply []ast.FileObject
+
+		parseOutputs []fsfake.ParserOutputs
 	}{
 		{
 			name:   "no objects",
 			format: filesystem.SourceFormatUnstructured,
+			parseOutputs: []fsfake.ParserOutputs{
+				{}, // One Parse call, no results or errors
+			},
 		},
 		{
 			name:              "implicit namespace if unstructured and not present",
 			format:            filesystem.SourceFormatUnstructured,
 			namespaceStrategy: configsync.NamespaceStrategyImplicit,
-			parsed: []ast.FileObject{
-				fake.Role(core.Namespace("foo")),
+			parseOutputs: []fsfake.ParserOutputs{
+				{
+					FileObjects: []ast.FileObject{
+						fake.Role(core.Namespace("foo")),
+					},
+				},
 			},
-			want: []ast.FileObject{
-				fake.UnstructuredAtPath(kinds.Namespace(),
-					"",
-					core.Name("foo"),
+			expectedObjsToApply: []ast.FileObject{
```

Comment:
These look more changed that they are. I just re-ordered them to match the new consistent ordering, with added implicit namespaces last.

Reply:
I also updated the `SyncTokenAnnotationKey` annotation value to match the commit added by using a populated `state.cache.source` as input.